### ISSUE TITLE
In an UWP application the change of language does not work well!

### ISSUE THREAD

**Issue by vsfeedback** [microsoft/microsoft-ui-xaml#6318]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/In-an-UWP-application-the-change-of-addr/1581878)._

---
[severity:It's more difficult to complete my work]
Hello,

I have several UWP applications that are in three languages : en-US es-ES and fr-FR.

I have developed a simple language change interface that requires a language selection and does:

                string cult = "en-US" ; // for exemple 
                ApplicationLanguages.PrimaryLanguageOverride = cult;
                CultureInfo culture = new CultureInfo(cult);
                CultureInfo.DefaultThreadCurrentCulture   = culture;
                CultureInfo.CurrentCulture                = culture;
                Thread.CurrentThread.CurrentCulture       = culture;
                ResourceContext.GetForCurrentView().Reset();
                ResourceContext.GetForViewIndependentUse().Reset();

Before re-loading the homepage.

This works perfectly in development mode, and I can change language and culture between the 3 proposed.

But once the application was delivered and released by Microsoft, if I install one of this applications from the Store: the language change works badly!

I can change language between en-US and fr-FR but impossible to get es-ES!

I tested different code modifications, for example the addition of :

await Task.Delay(100);

at the end of language modification but it does not solve the problem.

Do you have an idea of the origin of the problem?

Thank you in advance for your help.

Regards 
COM-ADD

---
#### COM-ADD on 11/13/2021, 08:49 AM: 

<p>Mistake: it must be understood that the change of LANGUAGE does not work!</p>

#### Feedback Bot on 11/15/2021, 01:50 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by StephenLPeters** [comment]
@MikeHillberg thoughts or where to route?

**Comment by thiago-cs** [comment]
Same here. I am working with **en-US**, **es-ES**, and **pt-BR**.
During development everything works great, but once installed from the store or sideloaded, I cannot get **es-ES** to work on my pc.
I installed my app from the store on a virtual machine and only **en-US** works in that VM (i.e. not even **pt-BR** works).
I am running Win10 Pro (fully updated) on both machines.

**Comment by thiago-cs** [comment]
So, apparently the issue has to do with selective installation where Windows installs only the app language resources for which there is a corresponding language package installed on Windows.

If I install any Spanish language package on Windows and reinstall my app, my **es-ES** localization works.

Alternatively, there is a [step-by-step](https://docs.microsoft.com/en-us/previous-versions/dn482043(v=vs.140)) from Microsoft docs that shows how to ensure that user-interface strings in multiple languages are installed with your app, not just the language that matches the user language preference of the device. That step-by-step is for _Windows 8.1 UWP_ but worked for my _Windows 10 UWP_.

Basically you just have to add 2 XML configuration files in the main project.

- One named `priconfig.default.xml`, with the list of your supported languages:
```
<default>
	<qualifier name="Language" value="en;es;pt" />
</default>
```
- The other one named `priconfig.packaging.xml`:
```
<packaging>
	<autoResourcePackage qualifier="Scale" />
	<autoResourcePackage qualifier="DXFeatureLevel" />
</packaging>
```
I just updated my app in the store and can confirm that following those steps **did solve** this issue for me.
